crates/backup-cli/src/adapters/contract_interface.rs

```diff
@@ -0,0 +1,81 @@
+use std::path::{Path, PathBuf};
+
+use ed25519_dalek::VerifyingKey;
+use mpc_contract::state::ProtocolContractState;
+use tokio::fs::File;
+use tokio::io::AsyncReadExt;
+
+use crate::ports::ContractInterface;
+
+const CONTRACT_STATE_FILENAME: &str = "contract_state.json";
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("failed to open file: {0}")]
+    OpenFile(tokio::io::Error),
+
+    #[error("could not read from file: {0}")]
+    Read(tokio::io::Error),
+
+    #[error("failed to deserialize secrets")]
+    JsonDeserialization(serde_json::Error),
+}
+
+pub struct SimpleContractInterface {
+    contract_state_path: PathBuf,
+}
```

Comment:
When do we need this type? I guess this is just a mock for tests, and not intended to be the "real" implementation right? In this case I'd expect this to be implemented in a test module or test helper module.